syntax = "proto3";

package opencrs;

service ExploitService {
    rpc Exploit(exploitationRequest) returns (exploit) {}
}

// Request
message exploitationRequest {
    // Binary to exploit
    bytes binary = 1;

    // Input stream to use. The string representation of the InputStreams enumeration.
    string input_stream = 2;

    // Boolean indicating if only the overflow attacks are used
    bool overflow_only = 3;

    // Boolean indicating if only the format string attacks are used
    bool format_only = 4;

    // Names of win functions, separated with commas
    string serialized_win_funcs = 5;
}

// Response
message exploit {
    // Pickled ZeratoolExploit object
    bytes pickledExploit = 1;
}